streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -714,6 +727,32 @@ public void close(boolean clean,
         taskClosed = true;
     }
 
+    private void retrieveCommittedTimestamp(final TopicPartition partition) {
+        final OffsetAndMetadata metadata = consumer.committed(partition);
+
+        if (metadata != null && metadata.metadata().length() != 0) {
```

Comment:
Maybe refactor out a helper for the above condition?

Reply:
Yeah, it looks like it appeared several times in the code, might want to add it as some separate static helper method.